pytentiostat/main.py

```diff
@@ -3,19 +3,68 @@
 from pytentiostat.reporter import save_data_to_file
 from pytentiostat.tester import experiment
 from pytentiostat.routines import startup_routine, closing_routine
+from pytentiostat.plotter import plot_updater, plot_initializer
+import matplotlib.pyplot as plt
+import sys
+import time
+
+
+class BoardCom:
+    def __init__(self):
+        self.com, self.board, self.pin_a0, self.pin_a2, self.pin_d9 = startup_routine()
+
+    def configure_board(self):
+        self.com, self.board, self.pin_a0, self.pin_a2, self.pin_d9 = startup_routine()
+
 
 config_data = parse_config_file()
 
-# Initialize Experiment
-com, board, a0, a2, d9 = startup_routine()
+try:
+    # Initialize Experiment
+    board_instance = BoardCom()
+    # com, board, a0, a2, d9 = startup_routine()
+except KeyboardInterrupt:
+    sys.exit("Stopped. Make any changes and then restart the software.\nExiting...")
+
+board_objects = (board_instance.pin_a0, board_instance.pin_a2, board_instance.pin_d9)
+
+reconfig = "sting"
+
+while True:
+    try:
+        # Run the experiment and get the config_data
+        times, voltages, currents = experiment(config_data, *board_objects)
+        break
+    except KeyboardInterrupt:
+        reconfig = input("Experiment stopped. To restart the same experiment, press Enter. \n"
+                         "If you want to do a different experiment, type \"New\"\n"
+                         "If you need to reconnect the poteniostat, type \"Reconnect\" \n")
+        if reconfig == "":
+            reconfig = "string"
+            plt.close()
+            continue
+        elif reconfig.lower() == "new":
+            reconfig = "string"
+            plt.close()
+            config_data = parse_config_file()
+            continue
+        elif reconfig.lower() == "reconnect":
+            reconfig = "string"
+            plt.close()
+            closing_routine(board_instance.board, board_instance.pin_d9)
+            board_instance.configure_board()
+            board_objects = (board_instance.pin_a0, board_instance.pin_a2, board_instance.pin_d9)
+            continue
+        elif reconfig == "string":
```

Comment:
Is this statement ever seen? I think you can remove it but you should test the behavior.

You probably do need an else statement to capture what happens if something other than nothing, "new", or "reconnect" are typed.  Or better, allow any string that is not new or reconnect to be considered as an abort.  Do this by having the "new" and "reconnect" logics first, then an else with the logic currently in ""

Reply:
I see that I need to do something about the else case. I now use a verify_input() function to force the user to input a valid string. Let me know what you think.

Reply:
I think that is overkill.  I would just assume everything is an abort unless something is typed that is a valid non-abort.

Reply:
Now it checks for "reconnect" or "new" and if anything else is entered, it aborts but asks the user if they want to save first. After aborting, it loops back up and says "Press enter to start the experiment".